sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -916,7 +916,7 @@ object ArraySortLike {
     _FUNC_(array[, ascendingOrder]) - Sorts the input array in ascending or descending order
       according to the natural ordering of the array elements. Null elements will be placed
       at the beginning of the returned array in ascending order or at the end of the returned
-      array in descending order.
+      array in descending order. NaN is greater than any non-NaN value for DoubleType/FloatType.
```

Comment:
```suggestion
      array in descending order. NaN is greater than any non-NaN value for double/float type.
```

Reply:
I think we should mention NaN before null.

Reply:
> I think we should mention NaN before null.

Done